backend/src/test/java/com/woowacourse/f12/application/product/ProductServiceTest.java

```diff
@@ -161,4 +114,25 @@ class ProductServiceTest {
                                 ETC_CONSTANT, 0.5))
         );
     }
+
+    @Test
+    void 제품명과_카테고리로_제품_목록을_조회한다() {
+        // given
+        Product product = KEYBOARD_1.생성(1L);
+        Pageable pageable = PageRequest.of(0, 2);
+        given(productRepository.findBySearchConditions("키보드1", KEYBOARD, pageable))
```

Comment:
```suggestion
        given(productRepository.findBySearchConditions(product.getName(), product.getCategory(), pageable))
```
product 를 활용해보는건 어떨까요?

Reply:
`findBySearchConditions`에는 사용자가 검색하려는 제품명과 카테고리를 전달하는 부분이라, `product`의 값을 그대로 사용하는 것은 의도와 맞지 않을 것 같아요!
차라리 ProductSearchRequest의 query와 category를 사용하는게 의도에 적합할 것 같은데 어떻게 생각하시나요??
```java
ProductSearchRequest productSearchRequest = new ProductSearchRequest("키보드1", KEYBOARD_CONSTANT);
given(productRepository.findBySearchConditions(productSearchRequest.getQuery(),
                productSearchRequest.getCategory().toCategory(), pageable))
                .willReturn(new SliceImpl<>(List.of(product), pageable, false));
```

Reply:
네.